tremor-script/src/lib.rs

```diff
@@ -25,6 +25,8 @@
 )]
 // TODO We need this for simd-json-derive
 #![allow(clippy::missing_errors_doc)]
+// often does not work because borrow checker
+#![allow(clippy::option_if_let_else)]
```

Comment:
Can we put this around the cases where the borrow check fails not as a generic over the whole file?